src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2142,10 +2142,27 @@ private void AddCustomAttributesToTable(IEnumerable<TypeReferenceWithAttributes>
 
         private void AddCustomAttributeToTable(EntityHandle parentHandle, ICustomAttribute customAttribute)
         {
-            metadata.AddCustomAttribute(
-                parent: parentHandle,
-                constructor: GetCustomAttributeTypeCodedIndex(customAttribute.Constructor(Context)),
-                value: GetCustomAttributeSignatureIndex(customAttribute));
+            IMethodReference constructor = customAttribute.Constructor(Context);
```

Comment:
>customAttribute.Constructor(Context) [](start = 43, length = 36)

This PR make way more changes than I would expect. I thought when we discussed this issue offline, the suggestion was to perform the check in the ```customAttribute.Constructor(Context)``` call. Since the synthesized default struct constructor is not a real method, it should not be possible to get IMethodReference for it.

Reply:
@AlekseyTS there are a few places in the codebase that don't expect null to be returned from `Constructor()`. I'm investigating further.

---
In reply to: [188084807](https://github.com/dotnet/roslyn/pull/26815#discussion_r188084807) [](ancestors = 188084807)